refactor(formatter): Expose explicit detect_code_removal api

leaysgur opened the pull request:
- Before: `Formatter::new().build()` panics when detected
- After: Need to use `detect_code_removal()` api explicitly

It has also been adapted to use `oxfmt`.

This change allows `monitor-oxc` to retain the output of existing `FormatterRunner` while also enabling separate results for `FormatterDCRRunner`.

leaysgur commented:
* **#15619** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15619?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15619?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15619).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Nov 12, 6:16 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 12, 6:16 AM UTC**: `Dunqing` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

leaysgur committed:
refactor(formatter): Expose explicit detect_code_removal api (#15619)

- Before: `Formatter::new().build()` panics when detected
- After: Need to use `detect_code_removal()` api explicitly

It has also been adapted to use `oxfmt`.

This change allows `monitor-oxc` to retain the output of existing `FormatterRunner` while also enabling separate results for `FormatterDCRRunner`.
